### ISSUE TITLE
Subnet overlapping

### ISSUE THREAD

**Issue by grex-gh** [telepresenceio/telepresence#3249]
**Describe the bug**

With telepresence 2.13.0 and above we are getting errors with overlapping subnets. Downgrading clients telepresence version to v2.12.2 works even if server/traffic-manager is on the latest v2.14.0.

When running `telepresence connect` we get this error:
```
telepresence connect: error: connector.Connect: subnet 10.8.32.0/20 overlaps with existing route "10.8.0.0/13 via 172.16.21.33 dev tun0, gw 172.16.21.1". Please see https://www.getambassador.io/docs/telepresence/latest/reference/vpn for more information
```

**Expected behavior**

With version prior to v2.13.0 running same command:

`telepresence connect`
```
Launching Telepresence User Daemon
Launching Telepresence Root Daemon
Connected to context dev (https://xx.xx.xx.xx)
```

**Comment by knlambert** [comment]
Hey @grex-gh, did you notice the documentation indicated in the error message ? It references this [page](https://www.getambassador.io/docs/telepresence/latest/reference/vpn) of the documentation which explains the issue.

We changed the behavior between 2.13 and 2.14: in the past, when you were having a subnet conflict, Telepresence wasn't saying anything, which could lead to other issues.

It's not failing silently anymore. If this conflict is normal for you, you can workaround it by upgrading the traffic-manager setup with this kind of command:

```cli
telepresence helm upgrade --set client.routing.allowConflictingSubnets={cird1,cidr2}
```

**Comment by wrbbz** [comment]
So, if I got it right:
If our k8s subnet is 10.8.0.0/13, our home subnet is 10.8.32.0/20, our colleague home subnet is 10.10.10.0/24 etc., we have to add every cidr in traffic-manager configuration?
Sounds painful, imho

UPD:
We added the full subnet there (i.e. 10.8.0.0/13). Nevertheless, I do not get the possible problems

**Comment by fragpit** [comment]
but `telepresence helm upgrade --set client.routing.allowConflictingSubnets="{10.43.0.0/16}"` also fails with 
```
telepresence helm upgrade: error: subnet 10.43.0.0/16 overlaps with existing route "10.0.0.0/8 via 10.x.x.x dev xxx, gw 10.x.x.x". Please see https://www.getambassador.io/docs/telepresence/latest/reference/vpn for more information
```

**Comment by netofontes** [comment]
Same issue here. The option `client.routing.allowConflictingSubnets` isn't doing anything.
```
telepresence helm upgrade --set client.routing.allowConflictingSubnets="{10.43.0.0/16,8.0.0.0/5}"

telepresence helm upgrade: error: subnet 10.43.0.0/16 overlaps with existing route "8.0.0.0/5 via X.X.X.X dev utun3, gw X.X.X.X". Please see https://www.getambassador.io/docs/telepresence/latest/reference/vpn for more information
```
Versions:
```
Client         : v2.14.0
Root Daemon    : v2.14.0
User Daemon    : v2.14.0
Traffic Manager: not connected
```

**Comment by spudfkc** [comment]
I was experiencing the same issue where doing an `upgrade --set client.routing.allowConflictingSubnets={}` was not doing anything other than giving me the same error about overlapping subnets. This was happening when I was connected to my workplace's VPN.
E.x.

```sh
# Errors out for some reason
telepresence helm upgrade --set client.routing.allowConflictingSubnets='{10.96.0.0/12,10.1.6.0/24}'
telepresence helm upgrade: error: subnet 10.1.6.0/24 overlaps with existing route "10.0.0.0/8 via 10.21.244.28 dev utun3, gw 10.21.244.28". Please see https://www.getambassador.io/docs/telepresence/latest/reference/vpn for more information
```

I was able to workaround this by doing an uninstall, and then an install with the conflicting subnets set:

E.x.
```sh
telepresence helm uninstall
telepresence helm install --set client.routing.allowConflictingSubnets='{10.96.0.0/12,10.1.6.0/24}'

telepresence connect    # works!
```

**Comment by cindymullins-dw** [comment]
Thanks, @spudfkc, for confirming `allowConflictingSubnets` works on uninstall / install.  I'll label it as a bug with the -`helm upgrade` command for now.

**Comment by drewfish** [comment]
I've run into the same issue (but for 192.168/16). I tried `telepresence helm uninstall/install` with `--set "client.routing.neverProxySubnets={192.168.0.0/16}"` and also tried `telepresence connect --never-proxy 192.168.0.0/16` but still got the same error. I was really hoping the "never proxy" feature would prevent this conflict, since in fact I don't want 192.168 from the cluster.

I removed the `client.routing.neverProxySubnets` and changed it to `client.routing.allowConflictingSubnet={192.168.0.0/16}` and it works now, and `netstat -rn -f inet` shows a new `192.168.0/17` routed to `utun3` (presumed to be from Telepresence).

**Comment by cindymullins-dw** [comment]
@grex-gh , @wrbbz , @fragpit , @netofontes , wondering if this works for you as well?
```
telepresence helm uninstall
telepresence helm install --set client.routing.allowConflictingSubnets='{}'
```

**Comment by cyenxchen** [comment]
I encountered the same problem, as shown in the following image:
![image](https://github.com/telepresenceio/telepresence/assets/22564041/9c22a798-61f0-4eec-895b-98b381852913)

**Comment by Sehat1137** [comment]
Same problem

```
$ telepresence helm install --set client.routing.allowConflictingSubnets='{}'

Traffic Manager installed successfully
$ telepresence connect
telepresence connect: error: connector.Connect: subnet 10.96.0.0/12 overlaps with existing route "10.0.0.0/8 via 10.31.3.38 dev tun0, gw 10.31.0.1". Please see https://www.getambassador.io/docs/telepresence/latest/reference/vpn for more information

```

Telepresence doesn't want work with existing VPN, at first I must turn off the VPN, but I don`t wanna choice between VPN and telepresence

**Comment by thallgren** [comment]
I think this is closed by #3494. Feel free to reopen if the problem still exists.
